client/src/pages/RegisterPage/RegisterPage.js

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import { connect } from "react-redux";
+import PropTypes from "prop-types";
+import { Field, reduxForm } from "redux-form";
+import userRegister from "../../actions/userRegister";
+
+class RegisterPage extends React.Component {
+  handleSubmit = event => {
+    const {username, postCode, houseNo, phoneNo, email, password} = this.props
+    event.preventDefault();
+    this.props.userRegister({username, postCode, houseNo, phoneNo, email, password});
+  };
+
+  render() {
+    return (
+      <React.Fragment>
+        <h1>Register</h1>
+        <form onSubmit={this.handleSubmit}>
+          <Field component="input" type="text" name="username" />
+          <Field component="input" type="text" name="postCode" />
+          <Field component="input" type="number" name="houseNo" />
+          <Field component="input" type="text" name="phoneNo" />
+          <Field component="input" type="email" name="email" />
+          <Field component="input" type="password" name="password" />
+          <Field component="input" type="password" name="passwordConfirm" />
+          <button>Sign up</button>
+          <Link to="/login">Login</Link>
```

Comment:
This link is breaking your app 
`Uncaught TypeError: Cannot read property 'username' of undefined`

Reply:
Struggled to reproduce the error. The initial values in the comment below were put in place to prevent errors like this.

Reply:
the error happens when you go back to login from register page, currently, it is not working and I can't merge it in this state... I would suggest opening an issue if you don't want to work on fixing it at the moment
This might help https://stackoverflow.com/questions/30115324/pass-props-in-link-react-router